File: Introduction: theory and tasks/gcd.py

def gcd(a, b):
    while all((a, b)):
        if a >= b:
            a = a % b
        else:
            b = b % a
    return max(a, b)

assert(gcd(18, 35) == 1)
assert(gcd(14159572, 63967072) == 4)
print('Done!')

File: Introduction: theory and tasks/gcd.rs

fn gcd(mut a: u64, mut b: u64) -> u64 {
    while (a != 0) & (b != 0) {
        if a >= b {
            a = a % b;
        } else {
            b = b % a;
        }
    }
    std::cmp::max(a, b)
}


fn main() {
    assert_eq!(gcd(18, 35), 1);
    assert_eq!(gcd(14159572, 63967072), 4);
    println!("Done!")
}
